components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/MetricHandler.java

```diff
@@ -0,0 +1,480 @@
+                    servicePort = servicePort.substring((servicePort.lastIndexOf(':') + 1),
+                            servicePort.lastIndexOf(DELIMITER));
+                    serviceInvokePort = Integer.parseInt(servicePort);
+                }
+                if ((null != synCtx.getProperty(SynapseConstants.IS_INBOUND))) {
+                    String inboundEndpointName = synCtx.getProperty(MetricConstants.INBOUND_ENDPOINT_NAME).
+                            toString();
+                    incrementInboundEndpointErrorCount(inboundEndpointName);
+                    stopTimers(synCtx.getProperty
+                            (MetricConstants.INBOUND_ENDPOINT_LATENCY_TIMER), synCtx);
+                } else if (null != synCtx.getProperty(MetricConstants.SYNAPSE_REST_API) &&
+                        (serviceInvokePort != internalHttpApiPort)) {
+                    String context = axis2MessageContext.getProperty(MetricConstants.TRANSPORT_IN_URL).
+                            toString();
+                    String apiInvocationUrl = axis2MessageContext.getProperty(MetricConstants.SERVICE_PREFIX).
+                            toString() + context.replaceFirst(DELIMITER, EMPTY);
+                    String apiName = getApiName(context, synCtx);
+
+                    incrementApiErrorCount(apiName, apiInvocationUrl);
+                    stopTimers(synCtx.getProperty(MetricConstants.API_LATENCY_TIMER), synCtx);
+                }
+            }
+        }
+        synCtx.setProperty(SynapseConstants.IS_ERROR_COUNT_ALREADY_PROCESSED, true);
+        return true;
+    }
+
+    @Override
+    public boolean handleStopServer() {
+        String host = System.getProperty(MicroIntegratorBaseConstants.LOCAL_IP_ADDRESS);
+        String port = System.getProperty(MetricConstants.HTTP_PORT);
+        String javaVersion = System.getProperty(MetricConstants.JAVA_VERSION);
+        String javaHome = System.getProperty(MetricConstants.JAVA_HOME);
+
+        metricReporterInstance.serverUp(host, port, javaVersion, javaHome);
+        return true;
+    }
+
+    @Override
+    public boolean handleDeployArtifacts(String artifactName, String artifactType, String startTime) {
+        metricReporterInstance.serviceUp(artifactName, artifactType);
+        return true;
+    }
+
+    @Override
+    public boolean handleUnDeployArtifacts(String artifactName, String artifactType, String startTime) {
+
+        return true;
+    }
+
+    /**
+     * Start the timers to observe request latency.
+     *
+     * @param synCtx           The Synapse Message Context
+     * @param serviceName      The proxy/api/inbound endpoint name
+     * @param serviceType      The service type (proxy/api/inbound endpoint)
+     * @param apiInvocationUrl The api invocation url
+     */
+    private void startTimers(MessageContext synCtx, String serviceName, String serviceType, String apiInvocationUrl) {
+
+        if (null == synCtx.getProperty("HAS_EXECUTED_FLOW")) {
```

Comment:
from where did this constant came ? please add to some constant file and document properly.

Reply:
This null check was removed after code refactoring.